Doc/deprecations/pending-removal-in-3.17.rst

```diff
@@ -9,14 +9,27 @@ Pending removal in Python 3.17
     3.17. Users should use documented introspection helpers like :func:`typing.get_origin`
     and :func:`typing.get_args` instead of relying on private implementation details.
   - :class:`typing.ByteString`, deprecated since Python 3.9, is scheduled for removal in
-    Python 3.17. Prefer :class:`~collections.abc.Sequence` or
-    :class:`~collections.abc.Buffer`. For use in type annotations, prefer a union, like
-    ``bytes | bytearray``, or :class:`collections.abc.Buffer`.
+    Python 3.17.
+
+   ``ByteString`` was originally intended to be an abstract type that would serve as a
+    supertype of both :class:`bytes` and :class:`bytearray`, but its semantics were never
+    clearly specified, and it was never understood properly by type checkers. See
+    :pep:`PEP 688 <688#current-options>` for more details.
+
+    Prefer :class:`~collections.abc.Sequence` or :class:`~collections.abc.Buffer`. For
+    use in type annotations, prefer a union, like ``bytes | bytearray``, or
+    :class:`collections.abc.Buffer`.
     (Contributed by Shantanu Jain in :gh:`91896`.)
 
 * :mod:`collections.abc`:
```

Comment:
Someone else put `collections.abc` after `typing`. Please could you alphabetise?